app/actions/get_user_from_omniauth_action.rb

```diff
@@ -0,0 +1,32 @@
+# frozen_string_literal: true
+
+class GetUserFromOmniauthAction < ApplicationAction
+  option :access_token
+
+  private
+
+  def perform_implementation
+    data = access_token.info
+    user = User.find_by(email: data['email'])
+
+    if user.nil?
+      user = create_user(data)
```

Comment:
Can we avoid variable reassignment?

Reply:
Yes, but I don't see any advantages! In case when we have `create_account_for(user)`

Reply:
Opting not to reassign variables (under any circumstances) offers the benefit of producing code that is clear and easy to comprehend, as well as straightforward to maintain.

When frequent reassignment occurs, it can lead to several potential issues:

- The possibility of a typo in the variable name (such as 'usr' instead of 'user').
- The need to constantly recall the prior value, which can complicate code reviews and reading.
- Difficulty in determining the potential values a variable can hold and the specific circumstances under which it may change.

In general, it's considered a less advisable practice and is best avoided. It is generally recommended to adhere to the principle of one variable, one assignment, and refrain from multiple reassignments.

Reply:
Got it, thanks!